dna/pDna/pDna.go

```diff
@@ -62,3 +102,81 @@ func Entropy(base Float32Base) float64 {
 
 	return answer
 }
+
+// Diff returns the vector that is the difference between 2 pDNA bases' probability vectors
+func Diff(p Float32Base, q Float32Base) Float64Diff {
+	return Float64Diff{
+		A: float64(p.A - q.A),
+		C: float64(p.C - q.C),
+		G: float64(p.G - q.G),
+		T: float64(p.T - q.T),
+	}
+}
+
+// Mag returns the magnitude of a difference probability vector
+// Mag is a float64
+func Mag(d Float64Diff) float64 {
+	return math.Sqrt(math.Pow(d.A, 2) + math.Pow(d.C, 2) + math.Pow(d.G, 2) + math.Pow(d.T, 2))
```

Comment:
Multiplication is faster than power function, unless the compiler recognizes power 2 and does multiplication instead. Not changing code now for efficiency boost